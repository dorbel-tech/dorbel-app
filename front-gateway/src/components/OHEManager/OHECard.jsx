import React from 'react';
import { Table, Col, Button, Panel, Image, Dropdown, MenuItem } from 'react-bootstrap';
import EditOHEModal from './EditOHEModal';
import DeleteOHEModal from './DeleteOHEModal';
import autobind from 'react-autobind';

import './OHECard.scss';

class OHECard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    autobind(this);
  }

  showEditModal(show) {
    this.setState({ showEditModal: show });
  }

  showDeleteModal(show) {
    this.setState({ showDeleteModal: show });
  }

  renderOheMenu() {
    const { ohe, editable } = this.props;

    if (!editable) {
      return null;
    } else {
      return (
        <Dropdown id={ohe.id + '_ohe_action'}>
          <Dropdown.Toggle noCaret bsStyle="link">
            <i className="fa fa-ellipsis-v" />
          </Dropdown.Toggle>
          <Dropdown.Menu className="dropdown-menu-left">
            <MenuItem onClick={() => this.showEditModal(true)}>עריכת מועד ביקור</MenuItem>
            <MenuItem onClick={() => this.showDeleteModal(true)}>מחיקה</MenuItem>
          </Dropdown.Menu>
        </Dropdown>
      );
    }
  }

  renderRegistrations(registrations) {
    if (!registrations) { return null; }

    return (
      <Table fill className="vertical-middle ohe-card-user-table">
        <tbody>
          {registrations.map(registration => (
            <tr key={registration.id} className="ohe-card-user-table-row">
              <td className="ohe-card-user-image-cell"><Image src={registration.user.picture} circle /></td>
              <td className="ohe-card-user-name-cell">{registration.user.first_name} {registration.user.last_name}</td>
              <td className="ohe-card-user-status-cell">{registration.is_active ? 'מתכוון להגיע' : 'הגעה בוטלה'}</td>
              <td className="ohe-card-user-phone-cell"><a href={'tel:' + (registration.user.phone || ' ')}>{registration.user.phone || ' '}</a></td>
              <td className="ohe-card-user-links-cell">
                <Button href={registration.user.facebook_link} disabled={!registration.user.facebook_link} bsStyle="link" target="_blank">
                  <i className={'fa fa-2x fa-facebook-square ' + (registration.user.facebook_link ? '' : 'ohe-card-facebook')}></i>
                </Button>
              </td>
            </tr>
          ))}
        </tbody>
      </Table>
    );
  }

  render() {
    const { ohe } = this.props;
    const numberOfActiveRegistrations = ohe.registrations ? ohe.registrations.filter(r => r.is_active).length : 0;

    return (
      <div className="ohe-card-row">
        <Panel>
          <Col xs={6}>
            <i className="ohe-card-row-icon pull-right fa fa-calendar-o" aria-hidden="true"></i>
            <span className="ohe-card-row-date">{ohe.dateLabel} - {ohe.dayLabel + '\''}</span><br />
            <span className="ohe-card-row-time">{ohe.timeLabel}</span>
          </Col>
          <Col xs={6} className="ohe-card-row-reg-num-col">
            <span className={(numberOfActiveRegistrations === 0 ? 'ohe-no-visits' : '')}>נרשמים לביקור ({numberOfActiveRegistrations})</span>
          </Col>
          <div className="ohe-card-menu">
            {this.renderOheMenu()}
          </div>
          {this.renderRegistrations(ohe.registrations)}
        </Panel>
        <EditOHEModal ohe={ohe} show={this.state.showEditModal} onClose={() => this.showEditModal(false)} />
        <DeleteOHEModal ohe={ohe} show={this.state.showDeleteModal} onClose={() => this.showDeleteModal(false)} />
      </div>
    );
  }
}

OHECard.propTypes = {
  ohe: React.PropTypes.object.isRequired,
  editable: React.PropTypes.bool
};

export default OHECard;
