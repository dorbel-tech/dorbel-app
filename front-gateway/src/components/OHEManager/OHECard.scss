@import "../sharedStyles/bsvars.scss";

.ohe-card-row {
  padding-bottom: 10px;
  position: relative;

  .panel-body {
    box-shadow: 0 4px 2px -2px rgba(0, 0, 0, 0.1);
    padding-right: 8px !important;
  }

  .ohe-card-row-icon-container {
    padding-right: 0;
  }

  .ohe-card-user-image-cell > img {
    width: 50px;
  }

  .ohe-card-user-status-cell {
    color: #3cb878;
  }

  .ohe-card-row-icon {
    font-size: 20px;
    line-height: 33px;
    padding-left: 22px;
    text-align: center;
    width: 73px;
  }

  @media (min-width: $screen-sm-min) {
    .ohe-card-user-image-cell {
      width: 80px;
    }

    .ohe-card-user-name-cell {
      text-overflow: ellipsis;
      white-space: nowrap;
      width: 170px;
    }

    .ohe-card-user-status-cell {
      width: 100px;
    }

    .ohe-card-user-phone-cell {
      width: 140px;
    }
  }

  @media (max-width: $screen-xs-max) {
    .panel-body {
      padding: 15px 0;
    }

    .ohe-card-row-icon {
      padding-left: 0;
      width: 40px;
    }

    .ohe-card-row-date,
    .ohe-card-row-time {
      padding-right: 9px;
    }

    .ohe-card-row-reg-num-col {
      padding-left: 30px;
      padding-right: 0;
    }

    .ohe-card-user-table {
      .ohe-card-user-table-row {
        border-bottom: 1px solid $gray-lighter;
      }

      td {
        border: none;
        display: inline-block;
      }

      .ohe-card-user-image-cell > img {
        width: 40px;
      }

      .ohe-card-user-name-cell {
        padding-top: 6px;
        vertical-align: top;
      }

      .ohe-card-user-status-cell {
        float: left;
        padding-top: 6px;
        vertical-align: top;
      }

      .ohe-card-user-phone-cell {
        margin-top: 22px;
        position: absolute;
        right: 57px;
        vertical-align: bottom;
      }

      .ohe-card-user-links-cell {
        left: -12px;
        padding-top: 18px;
        position: absolute;
      }
    }
  }
}

.ohe-no-visits {
  color: #cecece;
}

.ohe-card-menu {
  left: 0;
  position: absolute;
  top: 10px;
}

.ohe-card-facebook {
  color: #979797;
}
