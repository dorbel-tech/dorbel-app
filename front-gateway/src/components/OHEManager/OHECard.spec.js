'use strict';
import { shallow } from 'enzyme';
import React from 'react';
import { Dropdown } from 'react-bootstrap';
import OHECard from './OHECard';

describe('OHECard', () => {
  let props;
  let shallowOHECard;

  const oheCard = () => {
    if (!shallowOHECard) {
      shallowOHECard = shallow(
        <OHECard {...props} />
      ).dive();
    }
    return shallowOHECard;
  };

  beforeEach(() => {
    props = {
      ohe: {},
      appProviders: {}
    };
    shallowOHECard = undefined;
  });

  it('should render correctly given default values', () => {
    const registrationsTextWrapper = oheCard().find('.ohe-card-row-reg-num-col > span.ohe-no-visits');
    const registrations = oheCard().find('.ohe-card-user-table-row');

    expect(registrationsTextWrapper.text()).toEqual('נרשמים לביקור (0)');
    expect(registrations.length).toEqual(0);
    expect(oheCard()).toMatchSnapshot();
  });

  it('should render with an ohe menu', () => {
    props.editable = true;
    props.ohe.id = 'fakeId';

    expect(oheCard().find(Dropdown).props().id).toEqual('fakeId_ohe_action');
    expect(oheCard()).toMatchSnapshot();
  });

  it('should render with a single registration', () => {
    const fakeUser = { picture: 'pic', first_name: 'fn', last_name: 'ln', tenant_profile: {} };
    props.ohe.registrations = [
      { id: 'fakeId1', is_active: true, user: fakeUser }
    ];

    const registrationsTextWrapper = oheCard().find('.ohe-card-row-reg-num-col > span');
    const registrations = oheCard().find('.ohe-card-user-table-row');

    expect(registrationsTextWrapper.text()).toEqual('נרשמים לביקור (1)');
    expect(registrations.length).toEqual(1);
    expect(oheCard()).toMatchSnapshot();
  });

  it('should render with multiple registrations', () => {
    const fakeUser1 = { picture: 'pic1', first_name: 'fn1', last_name: 'ln1', tenant_profile: {} };
    const fakeUser2 = { picture: 'pic2', first_name: 'fn2', last_name: 'ln2', phone: '123', tenant_profile: { facebook_url: 'fbLink2' } };
    const fakeUser3 = { picture: 'pic3', first_name: 'fn3', last_name: 'ln3', phone: '456', tenant_profile: { facebook_url: 'fbLink3' } };
    props.ohe.registrations = [
      { id: 'fakeId1', is_active: true, user: fakeUser1 },
      { id: 'fakeId2', is_active: true, user: fakeUser2 },
      { id: 'fakeId3', is_active: true, user: fakeUser3 }
    ];

    const registrationsTextWrapper = oheCard().find('.ohe-card-row-reg-num-col > span');
    const registrations = oheCard().find('.ohe-card-user-table-row');

    expect(registrationsTextWrapper.text()).toEqual('נרשמים לביקור (3)');
    expect(registrations.length).toEqual(3);
    expect(oheCard()).toMatchSnapshot();
  });
});
