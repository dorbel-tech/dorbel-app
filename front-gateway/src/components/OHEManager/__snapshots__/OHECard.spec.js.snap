exports[`OHECard should render correctly given default values 1`] = `
<div
  className="ohe-card-row">
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}>
    <Col
      bsClass="col"
      className="ohe-card-row-icon-container"
      componentClass="div"
      xs={8}>
      <i
        aria-hidden="true"
        className="ohe-card-row-icon pull-right fa fa-calendar-o" />
      <span
        className="ohe-card-row-date">
         - 
        undefined\'
      </span>
      <br />
      <span
        className="ohe-card-row-time" />
    </Col>
    <Col
      bsClass="col"
      className="ohe-card-row-reg-num-col"
      componentClass="div"
      xs={4}>
      <span
        className="ohe-no-visits">
        נרשמים לביקור (
        0
        )
      </span>
    </Col>
    <div
      className="ohe-card-menu" />
  </Panel>
  <inject-EditOHEModal-with-appProviders
    ohe={Object {}}
    onClose={[Function]} />
  <inject-DeleteOHEModal-with-appProviders
    ohe={Object {}}
    onClose={[Function]} />
</div>
`;

exports[`OHECard should render with a single registration 1`] = `
<div
  className="ohe-card-row">
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}>
    <Col
      bsClass="col"
      className="ohe-card-row-icon-container"
      componentClass="div"
      xs={8}>
      <i
        aria-hidden="true"
        className="ohe-card-row-icon pull-right fa fa-calendar-o" />
      <span
        className="ohe-card-row-date">
         - 
        undefined\'
      </span>
      <br />
      <span
        className="ohe-card-row-time" />
    </Col>
    <Col
      bsClass="col"
      className="ohe-card-row-reg-num-col"
      componentClass="div"
      xs={4}>
      <span
        className="">
        נרשמים לביקור (
        1
        )
      </span>
    </Col>
    <div
      className="ohe-card-menu" />
    <Table
      bordered={false}
      bsClass="table"
      className="vertical-middle ohe-card-user-table"
      condensed={false}
      fill={true}
      hover={false}
      responsive={false}
      striped={false}>
      <tbody>
        <tr
          className="ohe-card-user-table-row">
          <td
            className="ohe-card-user-image-cell">
            <Image
              bsClass="img"
              circle={true}
              responsive={false}
              rounded={false}
              src="pic"
              thumbnail={false} />
          </td>
          <td
            className="ohe-card-user-name-cell">
            fn
             
            ln
          </td>
          <td
            className="ohe-card-user-status-cell">
            מתכוון להגיע
          </td>
          <td
            className="ohe-card-user-phone-cell">
            <a
              href="tel: ">
               
            </a>
          </td>
          <td
            className="ohe-card-user-links-cell">
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="link"
              disabled={true}
              target="_blank">
              <i
                className="fa fa-2x fa-facebook-square ohe-card-facebook" />
            </Button>
          </td>
        </tr>
      </tbody>
    </Table>
  </Panel>
  <inject-EditOHEModal-with-appProviders
    ohe={
      Object {
        "registrations": Array [
          Object {
            "id": "fakeId1",
            "is_active": true,
            "user": Object {
              "first_name": "fn",
              "last_name": "ln",
              "picture": "pic",
            },
          },
        ],
      }
    }
    onClose={[Function]} />
  <inject-DeleteOHEModal-with-appProviders
    ohe={
      Object {
        "registrations": Array [
          Object {
            "id": "fakeId1",
            "is_active": true,
            "user": Object {
              "first_name": "fn",
              "last_name": "ln",
              "picture": "pic",
            },
          },
        ],
      }
    }
    onClose={[Function]} />
</div>
`;

exports[`OHECard should render with an ohe menu 1`] = `
<div
  className="ohe-card-row">
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}>
    <Col
      bsClass="col"
      className="ohe-card-row-icon-container"
      componentClass="div"
      xs={8}>
      <i
        aria-hidden="true"
        className="ohe-card-row-icon pull-right fa fa-calendar-o" />
      <span
        className="ohe-card-row-date">
         - 
        undefined\'
      </span>
      <br />
      <span
        className="ohe-card-row-time" />
    </Col>
    <Col
      bsClass="col"
      className="ohe-card-row-reg-num-col"
      componentClass="div"
      xs={4}>
      <span
        className="ohe-no-visits">
        נרשמים לביקור (
        0
        )
      </span>
    </Col>
    <div
      className="ohe-card-menu">
      <Uncontrolled(Dropdown)
        id="fakeId_ohe_action">
        <DropdownToggle
          bsClass="dropdown-toggle"
          bsRole="toggle"
          bsStyle="link"
          noCaret={true}
          open={false}
          useAnchor={false}>
          <i
            className="fa fa-ellipsis-v" />
        </DropdownToggle>
        <DropdownMenu
          bsClass="dropdown-menu"
          bsRole="menu"
          className="dropdown-menu-left"
          pullRight={false}>
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            onClick={[Function]}>
            עריכת מועד ביקור
          </MenuItem>
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            onClick={[Function]}>
            מחיקה
          </MenuItem>
        </DropdownMenu>
      </Uncontrolled(Dropdown)>
    </div>
  </Panel>
  <inject-EditOHEModal-with-appProviders
    ohe={
      Object {
        "id": "fakeId",
      }
    }
    onClose={[Function]} />
  <inject-DeleteOHEModal-with-appProviders
    ohe={
      Object {
        "id": "fakeId",
      }
    }
    onClose={[Function]} />
</div>
`;

exports[`OHECard should render with multiple registrations 1`] = `
<div
  className="ohe-card-row">
  <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}>
    <Col
      bsClass="col"
      className="ohe-card-row-icon-container"
      componentClass="div"
      xs={8}>
      <i
        aria-hidden="true"
        className="ohe-card-row-icon pull-right fa fa-calendar-o" />
      <span
        className="ohe-card-row-date">
         - 
        undefined\'
      </span>
      <br />
      <span
        className="ohe-card-row-time" />
    </Col>
    <Col
      bsClass="col"
      className="ohe-card-row-reg-num-col"
      componentClass="div"
      xs={4}>
      <span
        className="">
        נרשמים לביקור (
        3
        )
      </span>
    </Col>
    <div
      className="ohe-card-menu" />
    <Table
      bordered={false}
      bsClass="table"
      className="vertical-middle ohe-card-user-table"
      condensed={false}
      fill={true}
      hover={false}
      responsive={false}
      striped={false}>
      <tbody>
        <tr
          className="ohe-card-user-table-row">
          <td
            className="ohe-card-user-image-cell">
            <Image
              bsClass="img"
              circle={true}
              responsive={false}
              rounded={false}
              src="pic1"
              thumbnail={false} />
          </td>
          <td
            className="ohe-card-user-name-cell">
            fn1
             
            ln1
          </td>
          <td
            className="ohe-card-user-status-cell">
            מתכוון להגיע
          </td>
          <td
            className="ohe-card-user-phone-cell">
            <a
              href="tel: ">
               
            </a>
          </td>
          <td
            className="ohe-card-user-links-cell">
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="link"
              disabled={true}
              target="_blank">
              <i
                className="fa fa-2x fa-facebook-square ohe-card-facebook" />
            </Button>
          </td>
        </tr>
        <tr
          className="ohe-card-user-table-row">
          <td
            className="ohe-card-user-image-cell">
            <Image
              bsClass="img"
              circle={true}
              responsive={false}
              rounded={false}
              src="pic2"
              thumbnail={false} />
          </td>
          <td
            className="ohe-card-user-name-cell">
            fn2
             
            ln2
          </td>
          <td
            className="ohe-card-user-status-cell">
            מתכוון להגיע
          </td>
          <td
            className="ohe-card-user-phone-cell">
            <a
              href="tel:123">
              123
            </a>
          </td>
          <td
            className="ohe-card-user-links-cell">
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="link"
              disabled={false}
              href="fbLink2"
              target="_blank">
              <i
                className="fa fa-2x fa-facebook-square " />
            </Button>
          </td>
        </tr>
        <tr
          className="ohe-card-user-table-row">
          <td
            className="ohe-card-user-image-cell">
            <Image
              bsClass="img"
              circle={true}
              responsive={false}
              rounded={false}
              src="pic3"
              thumbnail={false} />
          </td>
          <td
            className="ohe-card-user-name-cell">
            fn3
             
            ln3
          </td>
          <td
            className="ohe-card-user-status-cell">
            מתכוון להגיע
          </td>
          <td
            className="ohe-card-user-phone-cell">
            <a
              href="tel:456">
              456
            </a>
          </td>
          <td
            className="ohe-card-user-links-cell">
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="link"
              disabled={false}
              href="fbLink3"
              target="_blank">
              <i
                className="fa fa-2x fa-facebook-square " />
            </Button>
          </td>
        </tr>
      </tbody>
    </Table>
  </Panel>
  <inject-EditOHEModal-with-appProviders
    ohe={
      Object {
        "registrations": Array [
          Object {
            "id": "fakeId1",
            "is_active": true,
            "user": Object {
              "first_name": "fn1",
              "last_name": "ln1",
              "picture": "pic1",
            },
          },
          Object {
            "id": "fakeId2",
            "is_active": true,
            "user": Object {
              "facebook_link": "fbLink2",
              "first_name": "fn2",
              "last_name": "ln2",
              "phone": "123",
              "picture": "pic2",
            },
          },
          Object {
            "id": "fakeId3",
            "is_active": true,
            "user": Object {
              "facebook_link": "fbLink3",
              "first_name": "fn3",
              "last_name": "ln3",
              "phone": "456",
              "picture": "pic3",
            },
          },
        ],
      }
    }
    onClose={[Function]} />
  <inject-DeleteOHEModal-with-appProviders
    ohe={
      Object {
        "registrations": Array [
          Object {
            "id": "fakeId1",
            "is_active": true,
            "user": Object {
              "first_name": "fn1",
              "last_name": "ln1",
              "picture": "pic1",
            },
          },
          Object {
            "id": "fakeId2",
            "is_active": true,
            "user": Object {
              "facebook_link": "fbLink2",
              "first_name": "fn2",
              "last_name": "ln2",
              "phone": "123",
              "picture": "pic2",
            },
          },
          Object {
            "id": "fakeId3",
            "is_active": true,
            "user": Object {
              "facebook_link": "fbLink3",
              "first_name": "fn3",
              "last_name": "ln3",
              "phone": "456",
              "picture": "pic3",
            },
          },
        ],
      }
    }
    onClose={[Function]} />
</div>
`;
